### PR TITLE
chore: update to nextjs v14

### PR DESCRIPTION
**Pull request by lithdew**
Update to latest NextJS v14.0 for examples and @tanstack/react-query-next-experimental.

### CODE REVIEW

File: packages/react-query-next-experimental/package.json [hunk 1]
```diff
@@ -45,14 +45,14 @@
...
-    "next": "^13.5.6",
+    "next": "^14.0.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-error-boundary": "^3.1.4"
   },
   "peerDependencies": {
     "@tanstack/react-query": "workspace:^",
-    "next": "^13.0.0",
+    "next": "^14.0.0",
```
Comment (TkDodo) [inline packages/react-query-next-experimental/package.json:55]: I think it should be 13 || 14 ?

Comment (lithdew) [reply packages/react-query-next-experimental/package.json:55]: How about >=13.0.0?

**Comment by lithdew** [comment]
Updated peerDependency for @tanstack/react-query-next-experimental for NextJS to `>=13.0.0`.

Set default runtime for nextjs-suspense-streaming example to `edge` to mitigate a bug where prerendering route `/` attempts to fetch `http://localhost:3000/api/wait` on the `nodejs` runtime causes a prerender error at a build-time.

File: packages/react-query-next-experimental/package.json [hunk 2]
```diff
@@ -45,14 +45,14 @@
...
-    "next": "^13.5.6",
+    "next": "^14.0.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-error-boundary": "^3.1.4"
   },
   "peerDependencies": {
     "@tanstack/react-query": "workspace:^",
-    "next": "^13.0.0",
+    "next": ">=13.0.0",
```
Comment (TkDodo) [inline packages/react-query-next-experimental/package.json:55]: let's be explicit here please. If v15 comes out and this plugin isn't compatible anymore, it's better to fail installs.

Comment (lithdew) [reply packages/react-query-next-experimental/package.json]: Updated the peerDependency on next to `^13 || ^14`.

**Comment by TkDodo** [comment]
please fix the formatting

**Pull request merged**